Gopkg.toml

```diff
@@ -20,6 +20,11 @@ required = [
   source = "https://github.com/fsnotify/fsnotify.git"
   version="v1.4.7"
 
+# Override required for importing azure auth provider
+[[override]]
+name = "github.com/Azure/go-autorest"
+version="v10.15.5"
```

Comment:
We have been trying to avoid overrides because they bring an extra effort to ensure future compatibility. Is there a way to avoid it ?

Reply:
I've been trying to find a way to avoid this since I've started working with `client-go`, every time they update `client-go` to a new release, it seems to depend on the next version of `go-autorest`, but it always seems to end up being (very quickly) not the latest version, I think the package has similar release cycles.

The only way I can think of to avoid the import is to not import the Azure auth provider. When running `dep ensure` without the override it always chooses the latest package which isn't compatible with the `client-go` version imported into kubebuilder.

Unless you can think of a way that Dep can transitively read the `godeps.json`, I'm definitely open to ideas.

Reply:
>The only way I can think of to avoid the import is to not import the Azure auth provider. When running dep ensure without the override it always chooses the latest package which isn't compatible with the client-go version imported into kubebuilder.

One way to address it would be to document this by asking users to put this override in `Gopkg.toml` in a section which is not managed by Kubebuilder and not automatically doing it. 
Given that Go module support is getting matured, it will be a good idea to keep Gopkg.toml lightweight so that it will be easier to migrate to Go modules in future.

Reply:
> One way to address it would be to document this by asking users to put this override in `Gopkg.toml` in a section which is not managed by Kubebuilder and not automatically doing it.

Are you suggesting that rather than importing all auth providers, I modify the PR to import all but Azure, then add some separate documentation for those who need to auth against Azure? Explaining what they need to add and where?

Reply:
Yes, you are right. Re: where, a section (working with different cloud providers ?) in kubebuilder book may be helpful.